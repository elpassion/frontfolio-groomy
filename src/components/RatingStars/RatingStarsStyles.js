import styled from 'styled-components';

import { gaps } from 'styles/vars';

export const StarsRatingWrapper = styled.div`
  display: flex;
  width: auto;
  padding: 0 ${gaps.small} ${gaps.xsmall} 0;
`;

StarsRatingWrapper.displayName = 'StarsRatingWrapper';
