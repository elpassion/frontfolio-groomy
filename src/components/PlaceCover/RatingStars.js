import React from 'react';
import { number } from 'prop-types';
import { getReviewsRating } from '../../helpers/functions';
import RatingIcon from '../SvgAssets/RatingIcon';
import { StarsRatingWrapper } from './RatingStarsStyles';

class RatingStars extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      reviewsRating: getReviewsRating(this.props.placeId),
    };
  }

  render() {
    const { reviewsRating } = this.state;
    const maxRating = 5;

    return (
      reviewsRating > 0 && (
        <StarsRatingWrapper>
          {Array(maxRating)
            .fill(0)
            .map((element, index) => {
              return (
                <RatingIcon key={index} isActive={reviewsRating > index} />
              );
            })}
        </StarsRatingWrapper>
      )
    );
  }
}

RatingStars.propTypes = {
  placeId: number.isRequired,
};

export default RatingStars;
