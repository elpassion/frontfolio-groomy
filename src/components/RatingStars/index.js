import React from 'react';
import { number } from 'prop-types';

import { getReviewsRating } from 'data/api';
import { RatingIcon } from '../_svgAssets/RatingIcon';
import { StarsRatingWrapper } from './RatingStarsStyles';

const RatingStars = ({ venueId, maxRating }) => {
  const reviewsRating = getReviewsRating(venueId); // @todo: refactor to reuse on single review

  return (
    reviewsRating > 0 && (
      <StarsRatingWrapper>
        {Array(maxRating)
          .fill(0)
          .map((_element, index) => (
            <RatingIcon key={index} isActive={reviewsRating > index} />
          ))}
      </StarsRatingWrapper>
    )
  );
};

RatingStars.propTypes = {
  venueId: number,
};

RatingStars.defaultProps = {
  maxRating: 5,
};

RatingStars.displayName = 'RatingStars';
