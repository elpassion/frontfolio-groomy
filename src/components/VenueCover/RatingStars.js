import React from 'react';
import { number, object, bool } from 'prop-types';

import { getReviewsRating } from 'data/api';
import RatingIcon from '../_svgAssets/RatingIcon';
import { StarsRatingWrapper } from './RatingStarsStyles';

class RatingStars extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      reviewsRating: getReviewsRating(this.props.venueId), // @todo: refactor to reuse on single review
    };
  }

  render() {
    const { reviewsRating } = this.state;
    const maxRating = 5;

    return (
      reviewsRating > 0 && (
        <StarsRatingWrapper>
          {Array(maxRating)
            .fill(0)
            .map((element, index) => {
              return (
                <RatingIcon key={index} isActive={reviewsRating > index} />
              );
            })}
        </StarsRatingWrapper>
      )
    );
  }
}

RatingStars.propTypes = {
  isPlaceReview: bool,
  venueId: number,
  singleReview: object,
};

export default RatingStars;
