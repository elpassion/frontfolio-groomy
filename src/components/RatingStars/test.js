import React from 'react';
import { shallow, mount } from 'enzyme';

import RatingStars from './index';

it('renders correctly', () => {
  const wrapper = shallow(<RatingStars venueId={0} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly', () => {
  const wrapper = shallow(<RatingStars venueId={0} />);
  expect(wrapper).toMatchSnapshot();
});

describe('renders correct number of rating star icons', () => {
  it('with default maxRating', () => {
    const wrapper = mount(<RatingStars venueId={0} />);
    const expectedCount = wrapper.prop('maxRating');
    const actualCount = wrapper.find('RatingIcon').length;
    expect(actualCount).toEqual(expectedCount);
  });

  it('when maxRating is specified', () => {
    const wrapper = mount(<RatingStars venueId={0} maxRating={4} />);
    const expectedCount = wrapper.prop('maxRating');
    const actualCount = wrapper.find('RatingIcon').length;
    expect(actualCount).toEqual(expectedCount);
  });
});
