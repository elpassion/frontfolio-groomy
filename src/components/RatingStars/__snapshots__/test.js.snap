// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<StarsRatingWrapper>
  <RatingIcon
    isActive={true}
    key="0"
  />
  <RatingIcon
    isActive={false}
    key="1"
  />
  <RatingIcon
    isActive={false}
    key="2"
  />
  <RatingIcon
    isActive={false}
    key="3"
  />
  <RatingIcon
    isActive={false}
    key="4"
  />
</StarsRatingWrapper>
`;

exports[`renders correctly 2`] = `
<StarsRatingWrapper>
  <RatingIcon
    isActive={true}
    key="0"
  />
  <RatingIcon
    isActive={false}
    key="1"
  />
  <RatingIcon
    isActive={false}
    key="2"
  />
  <RatingIcon
    isActive={false}
    key="3"
  />
  <RatingIcon
    isActive={false}
    key="4"
  />
</StarsRatingWrapper>
`;
